libs/ui/catalog/src/lib/organisations-filter/organisations-filter.component.ts

```diff
@@ -33,8 +33,14 @@ export class OrganisationsSortComponent {
     },
   ]
   @Output() sortBy = new EventEmitter<SortByField>()
+  @Output() filterByValueChange = new EventEmitter<string>()
+  @Output() filterBy = this.filterByValueChange.pipe(debounceTime(300))
 
   selectOrderToDisplay(selectValue: string) {
     this.sortBy.emit(selectValue.split(',') as SortByField)
   }
+
+  filterOrganisations(inputValue: string) {
```

Comment:
Note that the children output is an Observable, which you could reach with a `@ChildView` component ref.
Parent output would just be `childrenOutput.pipe(debounce(300)`, no event handling with the component, no event emission, just propagation.

Reply:
Interesting idea! I guess I'd need an `ngOnInit` for this to update the ouput, though, no?

As doing the following, the `@Output()`'s initialization accesses too early the `searchInput`:
```
@ViewChild('search') searchInput
@Output() filterBy = this.searchInput.valueChange.pipe(debounceTime(300))
```

Reply:
Yes, the `ViewChild` are only accessible `afterViewInit`.